ctapipe/reco/HillasReconstructor.py

```diff
@@ -104,6 +104,7 @@ def __init__(self, config=None, parent=None, **kwargs):
         self.hillas_planes = {}
```

Comment:
While we are refactoring this class,  I suggest we make subarray the first argument of init and store it as self.subarray, similar to how all the other Components work (in particular TelescopeComponents). Then, you can remove subarray from the call to predict()

In fact, we should make `Reconstructor` a subclass of Component, so that this can be configured (even if there are no config options yet, just so it looks like other components).   

This could also be left to a future refactoring PR.

Reply:
I think this should be left to a dedicated PR since it would change the instantiation of `HillasReconstructor` and the call to `predict` in all the testing data and examples.